filesystem/node.go

```diff
@@ -167,3 +206,68 @@ func (n Node) Remove(_ context.Context, req *fuse.RemoveRequest) error {
 	// Unlink.
 	return n.cfs.remove(n.ID, req.Name, false /* !checkChildren */)
 }
+
+// Write writes data to 'n'. It may overwrite existing data, or grow it.
+func (n *Node) Write(_ context.Context, req *fuse.WriteRequest, resp *fuse.WriteResponse) error {
+	if n.IsDir {
+		return fuse.Errno(syscall.EISDIR)
+	}
+
+	n.mu.Lock()
+	defer n.mu.Unlock()
+
+	newLen := uint64(req.Offset) + uint64(len(req.Data))
+	if newLen > maxSize {
+		return fuse.Errno(syscall.EFBIG)
+	}
+
+	l := uint64(len(n.Data))
+	if newLen > l {
+		n.Data = append(n.Data, make([]byte, newLen-l)...)
+	}
+
+	written := copy(n.Data[req.Offset:], req.Data)
+	// TODO(marc): check that fuse forgets the node on errors, otherwise
+	// we'll have updated data in memory but not committed it.
+	if err := n.cfs.update(*n); err != nil {
```

Comment:
You're storing all of the file data in the inode! I guess that works, but ugh. I was thinking of a setup where the file data was stored in another table that looks like:

```
CREATE TABLE filesystem.filedata (
  id       INT,
  block INT,
  data  BYTES,
  PRIMARY KEY(id, block)
)
```

We'd then set the block size to something like 4K or 16K.